lib/facility.rb

```diff
@@ -1,14 +1,63 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
-  end
-
-  def add_services(service)
-    @services << service
-  end
-end
+
+    attr_reader :name,
+                :address, 
+                :phone, 
+                :services,
+                :registered_vehicles,
+                :collected_fees
+
+    def initialize(facility_info)
+        @name = facility_info[:name] 
+        @address = facility_info[:address]
+        @phone = facility_info[:phone]
+        @services = []
+        @registered_vehicles = []
+        @collected_fees = 0
+    end
+
+    # Add Service
+    def add_service(service)
+        @services << service
+    end
+
+    # Register a Vehicle
+    def register_vehicle(vehicle)
+        if @services.include? ("Vehicle Registration")
+            vehicle.change_registration_date
+            vehicle.update_plate_type
+            @collected_fees += vehicle.fees
+            @registered_vehicles << vehicle
+        else
+            return nil
+        end
```

Comment:
If you're just returning nil in the `else`, then you can just leave out the else and only have an if...end statement. Then if your code doesn't go down the `if` branch, it will automatically return nil.